mlx/ops.h

```diff
@@ -400,6 +400,14 @@ array where(
     const array& y,
     StreamOrDevice s = {});
 
+/** TODO */
+array nan_to_num(
+    const array& a,
+    const array& nan,
+    const array& posinf,
+    const array& neginf,
```

Comment:
I would also make these types just `float` for now. If we want to follow the Python API they could be `optional<float>`. Maybe that is better.. though slightly more complicated.